packages/service-library/src/servicelib/aiohttp/dev_error_logger.py

```diff
@@ -0,0 +1,36 @@
+
+from aiohttp.web import Application, middleware, Request, HTTPError
+from servicelib.aiohttp.typing_extension import Handler, Middleware
+import logging
+import traceback
+
+logger = logging.getLogger(__name__)
+
+_SEP = "|||"
+
+
+def _middleware_factory() -> Middleware:
+    @middleware
+    async def middleware_handler(request: Request, handler: Handler):
+        try:
+            return await handler(request)
+        except HTTPError as err:
+            fields = {
+                "Body": err.body,
+                "Status": err.status,
+                "Reason": err.reason,
+                "Headers": err.headers,
+                "Traceback": "\n".join(traceback.format_tb(err.__traceback__)),
+            }
+            formatted_error = "".join(
+                [f"\n{_SEP}{k}{_SEP}\n{v}" for k, v in fields.items()]
+            )
+            logger.debug("Error serialized to client:%s", formatted_error)
```

Comment:
question: as far as I see, this creates a text error. Could we not have a JSON formatted one like everywhere else?

Reply:
I have it formatted in an easy readable way. This is intended to be used for development/debugging. I'm not sure that having it formatted as json will help us.